new/resource-current-value.md

```diff
@@ -0,0 +1,98 @@
+---
+RFC: unassigned
+Author: John Keiser <jkeiser@chef.io
+Status: Draft
+Type: Standards Track
+---
+
+# Allow Current Value To Be Read From Resource
+
+Resources get a `current_value` method, which discovers the current value of
+the resource.
+
+## Motivation
+
+    As a Chef user,
+    I want to read the current value of certain resources (reading AWS object ids is a prime example),
+    so that I can do custom things in my recipes that the resource writers didn't think about.
+
+    As a Chef user,
+    I want to read the current value of certain resources,
+    So that I can make decisions in my recipes (such as only_if) based on actual data.
+
+    As a Chef resource developer,
+    I want to be able to expose all the read code I wrote to my users,
+    So I don't have to tell them to copy and paste it.
+
+## Specification
+
+All Resources get a method, `current_resource`, that returns the current value
+as a new resource. It does not cache this value. It returns `nil` when the
+resource does not exist. By default, it is implemented by creating the provider
+for one of the resource's actions, calling `load_current_resource` on that.
```

Comment:
Because the existing resource in the collection is a statement of what the user wants to be true (a desired value), not what is actually true (current_resource).